src/query/automaton_weight.rs

```diff
@@ -10,10 +10,14 @@ use crate::schema::{Field, IndexRecordOption};
 use crate::termdict::{TermDictionary, TermStreamer};
 use crate::{DocId, Score, TantivyError};
 
+use super::phrase_prefix_query::prefix_end;
+
 /// A weight struct for Fuzzy Term and Regex Queries
 pub struct AutomatonWeight<A> {
     field: Field,
     automaton: Arc<A>,
+    // Used to filter out terms that don't match the json path
+    json_path_bytes: Option<Vec<u8>>,
```

Comment:
It would be nice to use `Option<Box<[u8]>>` instead here to keep the size increase minimal when this is not used.